app/src/main/java/com/testfairy/TestFairyFeedbackOverlay.java

```diff
@@ -0,0 +1,614 @@
+		if (appToken == null) throw new AssertionError("TestFairy app token cannot be null.");
+
+		final Application a = (Application) context.getApplicationContext();
+		a.registerActivityLifecycleCallbacks(lifecycle);
+
+		visible = showImmediately;
+		visibleBeforePause = showImmediately;
+		token = appToken;
+		purpose = overlayPurpose;
+
+		uninstall = new Runnable() {
+			@Override
+			public void run() {
+				uninstall = null;
+				a.unregisterActivityLifecycleCallbacks(lifecycle);
+
+				if (instance != null) {
+					synchronized (instance) {
+						visibleBeforePause = false;
+						instance.removeOverlay(true);
+						instance = null;
+						beginCalled = false;
+					}
+				}
+			}
+		};
+
+		if (context instanceof Activity) {
+			final Activity activity = (Activity) context;
+			activity.runOnUiThread(new Runnable() {
+				@Override
+				public void run() {
+					onResume(activity);
+				}
+			});
+		}
+	}
+
+	/**
+	 * Call this if you no longer need the overlay during your app's lifetime.
+	 */
+	static public void uninstallOverlay() {
+		if (uninstall != null) {
+			uninstall.run();
+		}
+	}
+
+	/**
+	 * Toggles overlay visibility with animation if installed.
+	 */
+	static public void toggle() {
+		toggle(true);
+	}
+
+	/**
+	 * Toggles overlay visibility if installed.
+	 */
+	static public void toggle(boolean animated) {
+		if (instance != null && instance.container != null) {
+			if (visible) {
+				hide(animated);
+			} else {
+				show(animated);
+			}
+		}
+	}
+
+	/**
+	 * Shows overlay with animation if installed.
+	 */
+	static public void show() {
+		show(true);
+	}
+
+	/**
+	 * Shows overlay if installed.
+	 * @param animated Whether to run an animation.
+	 */
+	static public void show(boolean animated) {
+		if (instance != null && instance.container != null) {
+			final View container = instance.container;
+
+			if (animated) {
+				float widthHeight = dpToPx(instance.container.getContext(), ICON_SIZE_IN_DP);
+				TranslateAnimation translateAnimation = new TranslateAnimation(0, 0, widthHeight * 5, 0);
+				translateAnimation.setDuration(500);
+				translateAnimation.setAnimationListener(new Animation.AnimationListener() {
+					@Override
+					public void onAnimationStart(Animation animation) {
+						container.setVisibility(View.VISIBLE);
+					}
+
+					@Override
+					public void onAnimationEnd(Animation animation) {
+
+					}
+
+					@Override
+					public void onAnimationRepeat(Animation animation) {
+
+					}
+				});
+
+				container.startAnimation(translateAnimation);
+			} else {
+				container.setVisibility(View.VISIBLE);
+			}
+
+			visible = true;
+		} else {
+			visible = false;
+			visibleBeforePause = true;
+		}
+	}
+
+	/**
+	 * Hides overlay with animation if installed.
+	 */
+	static public void hide() {
+		hide(true);
+	}
+
+	/**
+	 * Hides overlay if installed.
+	 * @param animated Whether to run an animation.
+	 */
+	static public void hide(boolean animated) {
```

Comment:
No need to hide. If somebody said they want to provide feedback, they see the button. They can click on it, and then click cancel to dismiss. But there won't ever be a button on screen to remove the overlay we added.

Reply:
This means returning back from feedback form also uninstalls the overlay, right?